src/explore-education-statistics-common/src/components/form/rhf/__tests__/RHFForm.test.tsx

```diff
@@ -0,0 +1,491 @@
+                id="firstName"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.queryByText('Something went wrong')).toBeInTheDocument();
+    });
+
+    fireEvent.change(screen.getByLabelText('Firstname'), {
+      target: {
+        value: 'Another firstname',
+      },
+    });
+    await waitFor(() => {
+      expect(
+        screen.queryByText('Something went wrong'),
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  test('focuses error summary on submit', async () => {
+    render(
+      <FormProvider
+        initialValues={{
+          firstName: '',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().required(),
+        })}
+      >
+        {({ register }) => {
+          return (
+            <RHFForm id="test-form" showSubmitError onSubmit={noop}>
+              <input
+                id="test-form-firstName"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).toHaveFocus();
+    });
+  });
+
+  test('does not re-focus error summary when changing input', async () => {
+    render(
+      <FormProvider
+        initialValues={{
+          firstName: '',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().required(),
+        })}
+      >
+        {({ register }) => {
+          return (
+            <RHFForm id="test-form" showSubmitError onSubmit={noop}>
+              <label htmlFor="firstName">First name</label>
+              <input
+                id="firstName"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).toHaveFocus();
+    });
+    const input = screen.getByLabelText('First name');
+
+    input.focus();
+
+    fireEvent.change(input, {
+      target: {
+        value: 'a first name',
+      },
+    });
+    // await userEvent.type(screen.getByLabelText('First name'), 'a first name');
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).not.toHaveFocus();
```

Comment:
As earlier, using `getByRole` in a `waitFor`. We can optimise this with a `getByText`

Reply:
Yep and have realised that this test isn't correct - the alert has gone by the time it checks that it's not focused so will updat it here and in the original Form.test.tsx to make sure the alert is shown again but doesn't have focus.